posthog/cdp/templates/klaviyo/template_klaviyo.py

```diff
@@ -0,0 +1,264 @@
+from posthog.cdp.templates.hog_function_template import HogFunctionTemplate
+
+template_user: HogFunctionTemplate = HogFunctionTemplate(
+    status="beta",
+    id="template-klaviyo-user",
+    name="Klaviyo",
+    description="Updates a contact in Klaviyo",
+    icon_url="/static/services/klaviyo.png",
+    category=["Email Marketing"],
+    hog="""
+if (empty(inputs.externalId) and empty(inputs.email)) {
+    print('Email or External ID has to be set. Skipping...')
+    return
+}
+
+let body := {
+    'data': {
+        'type': 'profile',
+        'attributes': {
+            'location': {},
+            'properties': {},
+        }
+    }
+}
+
+if (not empty(person.properties.$geoip_latitude)) body.data.attributes.location.latitude := person.properties.$geoip_latitude
+if (not empty(person.properties.$geoip_longitude)) body.data.attributes.location.longitude := person.properties.$geoip_longitude
+if (not empty(person.properties.$geoip_city_name)) body.data.attributes.location.city := person.properties.$geoip_city_name
+if (not empty(person.properties.$geoip_country_name)) body.data.attributes.location.country := person.properties.$geoip_country_name
+if (not empty(person.properties.$geoip_continent_code)) body.data.attributes.location.region := person.properties.$geoip_continent_code
+if (not empty(person.properties.$geoip_postal_code)) body.data.attributes.location.zip := person.properties.$geoip_postal_code
+if (not empty(person.properties.$geoip_time_zone)) body.data.attributes.location.timezone := person.properties.$geoip_time_zone
+
+if (not empty(inputs.email)) body.data.attributes.email := inputs.email
+if (not empty(inputs.externalId)) body.data.attributes.external_id := inputs.externalId
```

Comment:
This could probably be condensed with some `let mapping := { 'latitude': '$geoip_latitude', ... }; for(let key, key in mapping) { ... }`, though this is likely fine too